libs/content-search-toolkit/src/services/elastic.service.ts

```diff
@@ -187,10 +190,17 @@ export class ElasticService {
     })
 
     const results = await Promise.all(requests)
-    return results.reduce((groupedResults, result, index) => {
-      groupedResults[metrics[index]] = result
-      return groupedResults
-    }, {})
+    return results.reduce(
```

Comment:
I think you can also put the result type as generic argument, which types the handler and second argument.

```
results.reduce<Record<string, RankEvaluationResponse<searchTermUnion>>>(...)
```

Also helps to define long types before, 

```
type RankResultMap = Record<string, RankEvaluationResponse<searchTermUnion>>
```

Reply:
Good point, thanks!